components/workshops.tsx

```diff
@@ -1,9 +1,80 @@
-const Workshops = () => (
-  <div className="bg-red-400">
-    <div className="flex flex-col sm:flex-row p-6 sm:p-12">
-      <h1 className="text-5xl sm:text-9xl font-bold">2. Workshops</h1>
+import WorkshopCard from "./card"
+import { useState, useEffect } from "react";
+import IEvent from "../utils/IEvent";
+
+const Workshops = () => {
+  const [events, setEvents] = useState<IEvent[]>([]);
+
+  const fetchEvents = async() => {
```

Comment:
I don't think this is a good way of fetching data. The problem with this is that there's a second or so where the "Workshops" section has no content while it's fetching the data. Next.js has a built-in way to solve this: [`getStaticProps()`](https://nextjs.org/docs/basic-features/data-fetching/get-static-props).

The way I would do this: Inside `pages/index.tsx`, create a new function at the bottom called `getStaticProps()` where you fetch from this API. Pass the data into the page, then make this `Workshops` component accept a prop called `events`, which you can then use directly inside the component. No client-side fetching or React hooks needed, plus the data will already be populated when the user visits the page!

I know this was probably not a very clear description, so plz let me know if I can help.